modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -687,6 +688,56 @@ public void handleUserDefinedType(
         }
     }
 
+    /** Using PostGIS table spatial_ref_sys to determine authority name and code of srid */
+    @Override
+    public CoordinateReferenceSystem createCRS(int srid, Connection cx) throws SQLException {
+        if (srid <= 0) {
+            return null;
+        }
+        try (Statement statement = cx.createStatement()) {
+            String sqlStatement =
+                    "SELECT AUTH_NAME, AUTH_SRID, SRTEXT FROM SPATIAL_REF_SYS WHERE SRID = "
+                            + Integer.toString(srid);
+            try (ResultSet result = statement.executeQuery(sqlStatement)) {
+                if (result.next()) {
```

Comment:
One more level of nesting can be avoided by flipping the condition:

```java
if (!result.next()) {
LOGGER.warning(
                            "SPATIAL_REF_SYS didn't have a row for srid: "
                                    + Integer.toString(srid));
return null;
}
```

Generally speaking, having only 100 columns, try to keep nesting to a minimum and eventually factoring out deeply nested bits to separate methods.

Reply:
Done